src/Map/MapRasterizer_P.cpp

```diff
@@ -514,32 +651,22 @@ void OsmAnd::MapRasterizer_P::rasterizePolyline(
     }
     else
     {
-        if (updatePaint(context, paint, primitive->evaluationResult, PaintValuesSet::Layer_minus2, false))
-            canvas.drawPath(path, paint);
-
-        if (updatePaint(context, paint, primitive->evaluationResult, PaintValuesSet::Layer_minus1, false))
-            canvas.drawPath(path, paint);
-
-        if (updatePaint(context, paint, primitive->evaluationResult, PaintValuesSet::Layer_0, false))
-            canvas.drawPath(path, paint);
-
-        if (updatePaint(context, paint, primitive->evaluationResult, PaintValuesSet::Layer_1, false))
-            canvas.drawPath(path, paint);
-
-        canvas.drawPath(path, paint);
-
-        if (updatePaint(context, paint, primitive->evaluationResult, PaintValuesSet::Layer_2, false))
-            canvas.drawPath(path, paint);
-
-        if (updatePaint(context, paint, primitive->evaluationResult, PaintValuesSet::Layer_3, false))
-            canvas.drawPath(path, paint);
-
-        if (updatePaint(context, paint, primitive->evaluationResult, PaintValuesSet::Layer_4, false))
-            canvas.drawPath(path, paint);
+        static const std::pair<PaintValuesSet, IMapStyle::ValueDefinitionId> layerRelatedIds[] =
```

Comment:
It's better to unwrap this loop

Reply:
Could you explain why it is better to unwrap?
Do you worried about multiple calls, some performance issue?
Or maybe I don't quite understand which cycle you were suggesting to unwrap. Could you please clarify this point?

During one of discussion were mentioned about simplification of code, improvements of readability (it's quite importent point in the scope of this project) etc., and avoid of code duplication, so we have to create functions for the duplicate piece of code. (BTW a few commits previously there was no loop here)

Reply:
My concern is of performance and needless refactoring

Reply:
Also, just noticed: https://www.mycompiler.io/view/8EWd9kw